packages/fetchai/connections/p2p_libp2p/libp2p_node/aea/api.go

```diff
@@ -360,19 +359,31 @@ func UnmarshalEnvelope(buf []byte) (*Envelope, error) {
 	return envelope, err
 }
 
-func (aea *AeaApi) listen_for_envelopes() {
+func (aea *AeaApi) listenForEnvelopes() {
 	//TOFIX(LR) add an exit strategy
 	for {
-		envel, err := read_envelope(aea.pipe)
-		if err != nil {
-			logger.Error().Str("err", err.Error()).Msg("while receiving envelope")
+		envel, err := HandleAcnMessageFromPipe(aea.pipe, aea, aea.AeaAddress())
+
+		var e *common.PipeError
+
+		if errors.As(err, &e) {
+			logger.Error().
+				Str("err", err.Error()).
+				Msg("pip error while receiving envelope. disconnect")
```

Comment:
`err` might be `!=nil` also when other bad things happen, i.e.:
- decoding error of ACN message
- decoding error of envelope
- unsupported ACN message